tests/foreman/cli/test_reporttemplates.py

```diff
@@ -63,878 +63,906 @@
+def test_positive_generate_hostpkgcompare(
+    request, local_org, local_ak, local_subscription, local_environment
+):
+    """Generate 'Host - compare content hosts packages' report
+
+    :id: 572fb387-86f2-40e2-b2df-e8ec26433610
+
+
+    :setup: Installed Satellite with Organization, Activation key,
+            Content View, Content Host, Subscriptions, and synced fake repo.
+
+    :steps:
+        1. hammer report-template generate --name 'Host - compare content hosts package' ...
+
+    :expectedresults: report is scheduled and generated containing all the expected information
+                      regarding host packages
+
+    :CaseImportance: Medium
+
+    :BZ: 1860430
+    """
+    # Add subscription to Satellite Tools repo to activation key
+    setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': local_org['id'],
+            'content-view-id': local_content_view['id'],
+            'lifecycle-environment-id': local_environment['id'],
+            'activationkey-id': local_ak['id'],
+        }
+    )
+
+    hosts = []
+    for _ in [0, 1]:
+        # Create VM and register content host
+        client = VirtualMachine(distro=DISTRO_RHEL7)
+        client.create()
+        request.addfinalizer(lambda: vm_cleanup(client))
+        client.install_katello_ca()
+        # Register content host, install katello-agent
+        client.register_contenthost(local_org['label'], local_ak['name'])
+        assert client.subscribed
+        hosts.append(Host.info({'name': client.hostname}))
+        client.enable_repo(REPOS['rhst7']['id'])
+        client.install_katello_agent()
+    hosts.sort(key=lambda host: host['name'])
+
+    host1, host2 = hosts
+    Host.package_install({'host-id': host1['id'], 'packages': FAKE_0_CUSTOM_PACKAGE_NAME})
+    Host.package_install({'host-id': host1['id'], 'packages': FAKE_1_CUSTOM_PACKAGE})
+    Host.package_install({'host-id': host2['id'], 'packages': FAKE_2_CUSTOM_PACKAGE})
+
+    result = ReportTemplate.generate(
+        {
+            'name': 'Host - compare content hosts packages',
+            'inputs': f'Host 1 = {host1["name"]}, ' f'Host 2 = {host2["name"]}',
+        }
+    )
+    result.remove('')
+
+    assert len(result) > 1
+    headers = f'Package,{host1["name"]},{host2["name"]},Architecture,Status'
+    assert headers == result[0]
+    items = [item.split(',') for item in result[1:]]
+    assert len(items)
+    for item in items:
+        assert len(item) == 5
+        name, host1version, host2version, arch, status = item
+        assert len(name)
+        assert (
+            (host1version == '-' and name in host2version)
+            or (name in host1version and host2version == '-')
+            or (name in host1version and name in host2version)
+        )
+        assert arch in ['x86_64', 'i686', 's390x', 'ppc64', 'ppc64le', 'noarch']
```

Comment:
The archs should have been defined in constants and then use it here.

Reply:
This is as it was in the testcase before the conversion, I'll take a look because you're right about the proper pattern for this.

Reply:
This is the only place in the framework where such a list of arches is used, and there's no constant defined for possible arch, so I'm going to leave this as-is.